corehq/apps/app_manager/suite_xml/post_process/workflow.py

```diff
@@ -330,25 +322,18 @@ def _frame_children_for_module(self, module, include_user_selections=True):
 
         return frame_children
 
-    def _get_first_command(self, module):
-        if module.module_type == "shadow":
-            module = module.source_module
```

Comment:
I don't see this log anywhere else. Is it not necessary?

My take is that because this is only used in places where `get_frame_children` is called with `module_only=True` the actual form command isn't appended to the result.

However, can we guarantee that the module datums for the shadow module will ALWAYS match those of the source module? 

```diff
   module_id = match.group(1)
-  module_datums = self.get_module_datums(module_id)
+  module_datums = self.get_module_datums(f'm{module.id}')
```
Previously the module ID was the ID of the 'source' module, now it's the ID of the shadow module.

Reply:
Wow, good find, thanks for flagging.

> My take is that because this is only used in places where get_frame_children is called with module_only=True the actual form command isn't appended to the result.

That was my interpretation and rationale for removing, yes.  You're right though, that the module specified in the command _is_ still relevant, though.  Here's the original:

```python
        match = re.search(r'^(m\d+)-(.*)', command)
        if not match:
            raise Exception("Unrecognized command: {}".format(command))
        target_module_id = match.group(1)
        target_form_id = match.group(2)

        frame_children = []

        module_command = id_strings.menu_id(target_module)
        module_datums = self.get_module_datums(target_module_id)
        form_datums = module_datums[target_form_id]
        frame_children = []
```
For shadow modules, `target_module` and `target_module_id` will refer to different modules.

> However, can we guarantee that the module datums for the shadow module will ALWAYS match those of the source module?

I'm certainly not confident in that (though I suppose that could be true).  I'm going to revert this PR and address this.